lib/pbench/test/unit/agent/task/test_move_results.py

```diff
@@ -1,15 +1,43 @@
-import os
 import datetime
+import logging
+import os
 import responses
 
+from pathlib import Path
 from pbench.cli.agent.commands.results import move_results
 from pbench.test.unit.agent.task.common import MockDatetime
 
 
+mdlog_tmpl = """[pbench]
+name = {}
+script = pbench-user-benchmark
+config = test-move-results
+date = YYYY-MM-DDTHH:MM:SS
+rpm-version = 0.00.0-1
+iterations = 1, 1
+
+[tools]
+hosts = agent.example.com
+group = default
+
+[tools/agent.example.com]
+sar = --interval=3
+
+[run]
+controller = agent.example.com
+start_run = YYYY-MM-DDTHH:MM:SS.000000000
+end_run = YYYY-MM-DDTHH:MM:SS.000000000
+
+[iterations/1]
+iteration_name = 1
+user_script = sleep
+"""
```

Comment:
This could be much better formatted, and maybe commented. The leading `"""` is not immediately obvious, and putting the first line of the config file on the same line doesn't help readability!

Reply:
> This could be much better formatted, and maybe commented.

How shall we format this differently?  Can we propose a formatting that does not differ from what the agent typically generates?

> The leading `"""` is not immediately obvious, and putting the first line of the config file on the same line doesn't help readability!

This is a template that is attempting to mimic the contents of an actual `metadata.log` file as constructed by the agent.  To that end, the first line of the config file has to be next to the `"""` to avoid an extra new line in the file, which is not how the agent constructs the file.  Further, the agent doesn't put comments in the file, and while that shouldn't be a problem, that is different from what the agent code does (though, maybe the agent should emit comments to help be understand the section contents).  The trailing `"""` has to be on a separate line so that the file will have a trailing newline.

I'll fix the `script`, `config`, and `date` fields to be consistent with the `run_name` variable.

Reply:
OK, I'll grant you the leading newline.

But I wasn't talking about a comment inside the config file quote, but rather before the assignment.

Reply:
Can this be added to unit test fixtures directory,? We do store the log files that we use for the unit tests in `fixtures` directory like `lib/pbench/test/unit/agent/fixtures/move_results/logs/metadata.log`

Reply:
> Can this be added to unit test fixtures directory,? We do store the log files that we use for the unit tests in `fixtures` directory like `lib/pbench/test/unit/agent/fixtures/move_results/logs/metadata.log`

Yes, another avenue to take would be to make a fixture of a mocked out pbench run directory with just a `metadata.log` file in it.  Then the test would just pass it to the `move_results` method.  I opted not to do that thinking that we'll want to write more tests to verify the behavior of `move_results`, where this methods allows us to taylor the contents a bit.